realm/realm-library/src/objectServer/java/io/realm/RealmFunctions.java

```diff
@@ -17,30 +17,173 @@
 
 import org.bson.codecs.configuration.CodecRegistry;
 
+import java.util.List;
+import java.util.concurrent.atomic.AtomicReference;
+
+import io.realm.internal.Util;
 import io.realm.internal.jni.JniBsonProtocol;
+import io.realm.internal.jni.OsJNIResultCallback;
+import io.realm.internal.objectstore.OsJavaNetworkTransport;
+
+/**
+ * A <i>Realm functions<i> manager to call MongoDB Realm functions.
+ *
+ * Arguments and results are encoded/decoded with the <i>Realm functions'</i> codec registry either
+ * inherited from the {@link RealmAppConfiguration#getDefaultCodecRegistry()} or set explicitly
+ * when creating the <i>Realm functions</i>-instance through {@link RealmUser#getFunctions(CodecRegistry)}
+ * or through the individual calls to {@link #callFunction(String, List, Class, CodecRegistry)}.
+ */
+// TODO Timeout is currently handled uniformly through OkHttpNetworkTransport configured through RealmAppConfig
+// FIXME Application wide invocation; like StitchAppClient appClient = Stitch.getDefaultAppClient();
```

Comment:
Not 100% sure what these FIXME's mean or if they are still relevant?

Reply:
Initially I though it was possible to execute a function without a user. Apparently it was due to implicitly resolving a _current user_. Since we want to avoid this implicit resolving of users this is no longer relevant.